druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java

```diff
@@ -1504,6 +1513,115 @@ public Void apply(ResultSet resultSet) {
             "\"upper\":\"12223\""));
   }
 
+  @Test public void testPushAggragateOnTime() {
+    String sql = "select \"product_id\", \"timestamp\" as \"time\" from \"foodmart\" where \"product_id\" = 1016 and "
+            + "\"timestamp\" < cast('1997-01-03' as timestamp) and \"timestamp\" > cast"
+            + "('1990-01-01' as timestamp)"
+            + " group by"
+            + "\"timestamp\", \"product_id\" ";
+
+    String druidQuery = "{\"queryType\":\"groupBy\",\"dataSource\":\"foodmart\","
+            + "\"granularity\":\"all\",\"dimensions\":[{\"type\":\"extraction\","
+            + "\"dimension\":\"__time\",\"outputName\":\"timestamp\","
+            + "\"extractionFn\":{\"type\":\"timeFormat\",\"format\":\"yyyy-MM-dd";
+    sql(sql).returnsUnordered("product_id=1016; time=1997-01-02 00:00:00").queryContains(druidChecker(druidQuery));
+  }
+
+
+  @Test public void testPushAggragateOnTimeWithExtractYear() {
+    String sql = "select EXTRACT( year from \"timestamp\") as \"year\",\"product_id\"  from \"foodmart\" where \"product_id\" = 1016 and "
+            + "\"timestamp\" < cast('1999-01-02' as timestamp) and \"timestamp\" > cast"
+            + "('1997-01-01' as timestamp)"
+            + " group by "
+            + " EXTRACT( year from \"timestamp\"), \"product_id\" ";
+
+    sql(sql).queryContains(
+            druidChecker(",\"granularity\":\"all\"", "{\"type\":\"extraction\","
+                    + "\"dimension\":\"__time\",\"outputName\":\"year\","
+                    + "\"extractionFn\":{\"type\":\"timeFormat\",\"format\":\"yyyy\","
+                    + "\"timeZone\":\"UTC\"}}")).returnsUnordered("year=1997; product_id=1016");
+  }
+
+
+  @Test public void testPushAggragateOnTimeWithExtractMonth() {
+    String sql = "select EXTRACT( month from \"timestamp\") as \"month\",\"product_id\"  from \"foodmart\" where \"product_id\" = 1016 and "
+            + "\"timestamp\" < cast('1997-06-02' as timestamp) and \"timestamp\" > cast"
+            + "('1997-01-01' as timestamp)"
+            + " group by "
+            + " EXTRACT( month from \"timestamp\"), \"product_id\" ";
+
+    sql(sql).queryContains(
+            druidChecker(",\"granularity\":\"all\"", "{\"type\":\"extraction\","
+                    + "\"dimension\":\"__time\",\"outputName\":\"monthOfYear\","
+                    + "\"extractionFn\":{\"type\":\"timeFormat\",\"format\":\"MM\","
+                    + "\"timeZone\":\"UTC\"}}"))
+            .returnsUnordered("month=01; product_id=1016", "month=02; product_id=1016",
+                    "month=03; product_id=1016", "month=04; product_id=1016", "month=05; product_id=1016"
+            );
+  }
+
+  @Test public void testPushAggragateOnTimeWithExtractDay() {
+    String sql = "select EXTRACT( day from \"timestamp\") as \"day\",\"product_id\"  from \"foodmart\" where \"product_id\" = 1016 and "
```

Comment:
We need to solve the problem with the _extraction result column_ name that you mentioned previously. I see that you are renaming the result of the _extract_, but that will not be enough. If we do not solve it but we check this patch in, this will result in incorrect results in those cases, right?